packages/grafana-o11y-ds-frontend/src/TemporaryAlert.tsx

```diff
@@ -30,22 +30,26 @@ const timeoutMap = {
 export const TemporaryAlert = (props: { severity: AlertVariant; text: string }) => {
   const style = getStyle(useTheme2());
   const [visible, setVisible] = useState(false);
+  const [timer, setTimer] = useState<NodeJS.Timeout>();
 
   useEffect(() => {
-    const timer = setTimeout(() => {
-      setVisible(false);
-    }, timeoutMap[props.severity]);
-
     return () => {
-      clearTimeout(timer);
+      if (timer) {
+        clearTimeout(timer);
+      }
     };
-  }, [props.severity, visible]);
+  }, [timer]);
 
   useEffect(() => {
     if (props.text !== '') {
       setVisible(true);
+
+      const timer = setTimeout(() => {
```

Comment:
I suspect you would have to clear the previous timer, if there is one, before creating this new timeout.